docs/sdg/knowledge-doc-ingestion.md

```diff
@@ -0,0 +1,96 @@
+
+## 2. Use Case
+
+To enable the ingestion and processing of a wide range of document types, this pipeline must handle formats including:
+
+- Markdown (MD)
+- PDF
+- TXT
+- ASCII Docs
+- DOCX
+- HTML
+- PPTX
+
+This proposal outlines how to build a pluggable system that accommodates these formats and integrates effectively into existing **SDG** and **RAG** workflows.
+
+## 3. Proposed Approach
+
+### 3.1 Custom InstructLab Schema Design
+
+- **Objective**: Define a custom "instructlab" schema that standardizes input formats for SDG and RAG pipelines. This schema will bridge the gap between various document types and the specific formats required for further processing.
+- **Modularity**: The system should allow easy extension, enabling support for new document types and processing workflows without disrupting core functionality.
+  
+The instructlab schema will serve as the intermediary format, supporting flexibility while ensuring compatibility with the ingestion process.
+
+### 3.2 PDF and Document Conversion via Docling
+
+- **Docling Integration**: We will leverage **Docling** to convert files into structured JSON, which will be the starting point for the instructlab schema. Individual components will post-process this JSON as per the requirements of the specific SDG and RAG workflows.
+  
+- **Docling v2**: We have collaborated with the Docling team to extend their tool’s capabilities, allowing conversion for PDF, HTML and DOCX formats. These new document types will be supported in the ingestion pipeline via docling's upgraded v2 release.
+
+### 3.3 Introducing the Document Chunking Command
+
+- **Command Overview**: We propose a new command, `ilab document format`, which will:
+  - Take a document path (defined in `qna.yaml`).
+  - Format and chunk the document into the desired schema.
+  
+- **Implementation Details**: Initially, this functionality will be integrated into the existing SDG repository. Over time, it can evolve into a standalone utility, allowing external integrations and wider usage.
+
+- **Example Workflow**:
+  ```bash
+  ilab document format --input path/to/document.pdf --output path/to/schema
+  ```
+  This command will ingest a document, process it into the instructlab schema, and output the result for further use in SDG or RAG workflows.
+
+### 3.4 Simplifying Git-Based Workflows for Users
+
+- **Current Challenge**: Knowledge documents are stored in Git-based repositories, which may be unfamiliar to many users.
+- **Proposed Solution**:
+  - Allow users to input a local directory and provide an automated script that:
```

Comment:
Rather than a "script," why not just have this be part of the code? We can detect if a given directory is git-tracked (e.g. by checking for a `.git` subdirectory) and do the manipulation described if not